### PR TITLE
MODORDERS-207 - Adding cases for checkin

### PR DESCRIPTION
**Pull request by KVupp**
## PURPOSE
Add new test cases to cover checkin flow

## APPROACH
- Added new util methods to create pieces and items as  they are not created by mod-orders during check in flow 
- Added new tests to cover scenarios 
    1) for checkin without items required in inventory(pieces will be created in a pre request step)
    2) checkin with items in inventory(pieces and items are created in pre-request step )
    3) revert checkin of pieces and also verifying that the item status and piece status are also reverted in inventory
    4) failure scenario which throws an error if a piece is tried to checkin again
    
Additional changes
1) Changed few util methods to cover scenarios for check-in where there can be holdings without items, and cleaning those records as well

### CODE REVIEW

**Comment by KVupp** [comment]
Folio-testing is being changed to proxy on http port but has issues running tests on port 443. Hence verified them on folio-snapshot
Screen Shot 2019-04-22 at 4 25 57 PM

**Comment by craigmcnally** [comment]
Kruthi, I'm getting failures running against folio-snapshot...  
image

I haven't looked yet but I'm guessing something isn't being cleaned up entirely

**Comment by KVupp** [comment]
I ran it back to back yesterday night and it was a success both the times 
Screen Shot 2019-04-22 at 10 31 59 PM
Screen Shot 2019-04-22 at 10 38 51 PM

Can you send me the error messages, I can look to see what's wrong

**Comment by piotr-kalashuk** [comment]
Hi @KVupp 
I also checked on fresh vagrant box and see the same failures

image

The error happens on item creation because `permanentLoanTypeId` is missing
image

File: mod-orders/mod-orders.postman_collection.json [hunk 1]
```diff
@@ -8251,85 +8373,549 @@
...
+								},
+								"description": "Create an order with minimal required fields (update based on [MODORDERS-136](https://issues.folio.org/browse/MODORDERS-136) and [MODORDERS-145](https://issues.folio.org/browse/MODORDERS-145)."
+							},
+							"response": []
+						}
+					],
+					"_postman_isSubFolder": true
+				},
+				{
+					"name": "check-in",
```
Comment (craigmcnally) [inline mod-orders/mod-orders.postman_collection.json]: This is a nit-pick but the can you make this "Check-in" for case consistency

**Comment by KVupp** [comment]
This PR sShould be all set now!
Also, I ran it with clean environment variables and everything passes.  Let me know if you see any other issue.
Screen Shot 2019-04-23 at 11 47 29 AM


P.S @craigmcnally  Also I was able to find the root cause for the new holdings failure that you were seeing. It was because one of the request was dependent on the create Inventory values and the snapshot had a value "Instance" for **Other**. I added a fix for that too.

**Comment by craigmcnally** [comment]
Looks good now.  Thanks Kruthi.  
image

**Pull request merged**